src/psyclone/psyir/nodes/node.py

```diff
@@ -1313,16 +1313,19 @@ def scope(self):
             f"Unable to find the scope of node '{self}' as none of its "
             f"ancestors are Container or Schedule nodes.")
 
-    def replace_with(self, node):
+    def replace_with(self, node, keep_name_in_context=True):
         '''Removes self, and its descendants, from the PSyIR tree to which it
         is connected, and replaces it with the supplied node (and its
         descendants).
 
         :param node: the node that will replace self in the PSyIR \
             tree.
         :type node: :py:class:`psyclone.psyir.nodes.node`
+        :param bool keep_name_in_context: whether to conserve the name \
+            referencing this node when it is within a parent with named nodes.
```

Comment:
I'm not sure it makes sense to mention parent here as that is an implementation? Perhaps just remove (when it is ... named nodes.)

Reply:
Done